### PR TITLE
Added cleanup-vpc-peering command to subctl->cloud

### PR DESCRIPTION
**Pull request by r2dedios**

### CODE REVIEW

File: pkg/subctl/cmd/cloud/vpcpeering/generic.go [hunk 1]
```diff
@@ -34,3 +34,18 @@ func newGenericVPCPeeringCommand() *cobra.Command {
...
+		Short: "Removes VPC peering between generic clusters for Submariner",
+		Long:  "This command create a vpc peering between two OCP cloud agnostic clusters",
+		Run:   cleanVpcPeerGenericCluster,
+	}
+
+	return cmd
+}
+
+func cleanVpcPeerGenericCluster(cmd *cobra.Command, args []string) {
+	// Nothing to peer here
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/generic.go]:
```suggestion
	// Nothing to clean up here
```

File: pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go [hunk 2]
```diff
@@ -23,15 +23,31 @@ import (
...
-var (
-	parentRestConfigProducer *restconfig.Producer
-)
+var parentRestConfigProducer *restconfig.Producer
 
 // NewCommand returns a new cobra.Command used to prepare a cloud infrastructure.
 func NewCommand(restConfigProducer *restconfig.Producer) *cobra.Command {
 	parentRestConfigProducer = restConfigProducer
+
+	// Create VPC-Peering
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go]:
```suggestion
```

File: pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go [hunk 3]
```diff
@@ -42,3 +58,20 @@ func NewCommand(restConfigProducer *restconfig.Producer) *cobra.Command {
 
 	return cmd
 }
+
+func newCleanupCommand(restConfigProducer *restconfig.Producer) *cobra.Command {
+	parentRestConfigProducer = restConfigProducer
+
+	// Clean VPC-Peering
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go]:
```suggestion
	// Clean up VPC-Peering
```

File: pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go [hunk 4]
```diff
@@ -42,3 +58,20 @@ func NewCommand(restConfigProducer *restconfig.Producer) *cobra.Command {
...
+	cmd := &cobra.Command{
+		Use:   "clean",
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go]:
```suggestion
		Use:   "cleanup",
```

File: pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go [hunk 5]
```diff
@@ -42,3 +58,20 @@ func NewCommand(restConfigProducer *restconfig.Producer) *cobra.Command {
...
+		Short: "Remove a VPC Peering between clusters",
+		Long:  `This command removes a VPC Peering between different clusters of the same cloud provider.`,
+	}
+
+	cmd.AddCommand(newCleanAWSVPCPeeringCommand())
+	cmd.AddCommand(newCleanGCPVPCPeeringCommand())
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go]:
```suggestion
	cmd.AddCommand(newCleanupGCPVPCPeeringCommand())
```

File: pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go [hunk 6]
```diff
@@ -42,3 +58,20 @@ func NewCommand(restConfigProducer *restconfig.Producer) *cobra.Command {
...
+	cmd.AddCommand(newCleanGenericVPCPeeringCommand())
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go]:
```suggestion
	cmd.AddCommand(newCleanupGenericVPCPeeringCommand())
```

File: pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go [hunk 7]
```diff
@@ -42,3 +58,20 @@ func NewCommand(restConfigProducer *restconfig.Producer) *cobra.Command {
...
+	parentRestConfigProducer = restConfigProducer
+
+	// Clean VPC-Peering
+	cmd := &cobra.Command{
+		Use:   "clean",
+		Short: "Remove a VPC Peering between clusters",
+		Long:  `This command removes a VPC Peering between different clusters of the same cloud provider.`,
+	}
+
+	cmd.AddCommand(newCleanAWSVPCPeeringCommand())
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/vpcpeering.go]:
```suggestion
	cmd.AddCommand(newCleanupAWSVPCPeeringCommand())
```

File: pkg/subctl/cmd/cloud/vpcpeering/gcp.go [hunk 8]
```diff
@@ -42,24 +39,65 @@ func newGCPVPCPeeringCommand() *cobra.Command {
...
+	}
+
+	gcp.ClientArgs.AddGCPFlags(cmd)
+	targetArgs.AddAWSFlags(cmd)
+
+	return cmd
+}
+
+// cleanVpcPeerGcp removes peering object and routes between two OCP clusters in AWS.
+func cleanVpcPeerGcp(cmd *cobra.Command, args []string) {
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/gcp.go]:
```suggestion
func cleanupVpcPeerGcp(cmd *cobra.Command, args []string) {
```

File: pkg/subctl/cmd/cloud/vpcpeering/gcp.go [hunk 9]
```diff
@@ -42,24 +39,65 @@ func newGCPVPCPeeringCommand() *cobra.Command {
...
+	}
+}
+
+// newCleanAWSVPCPeeringCommand removes a VPC Peering between different AWS clusters.
+func newCleanGCPVPCPeeringCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "gcp",
+		Short: "Removes VPC Peering on GCP cloud",
+		Long:  "This command cleans an OpenShift installer-provisioned infrastructure (IPI) on GCP cloud for Submariner uninstallation.",
+		Run:   cleanVpcPeerGcp,
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/gcp.go]:
```suggestion
		Run:   cleanupVpcPeerGcp,
```

File: pkg/subctl/cmd/cloud/vpcpeering/generic.go [hunk 10]
```diff
@@ -34,3 +34,18 @@ func newGenericVPCPeeringCommand() *cobra.Command {
 func vpcPeerGenericCluster(cmd *cobra.Command, args []string) {
 	// Nothing to peer here
 }
+
+func newCleanGenericVPCPeeringCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "generic",
+		Short: "Removes VPC peering between generic clusters for Submariner",
+		Long:  "This command create a vpc peering between two OCP cloud agnostic clusters",
+		Run:   cleanVpcPeerGenericCluster,
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/generic.go]:
```suggestion
		Run:   cleanupVpcPeerGenericCluster,
```

File: pkg/subctl/cmd/cloud/vpcpeering/generic.go [hunk 11]
```diff
@@ -34,3 +34,18 @@ func newGenericVPCPeeringCommand() *cobra.Command {
...
+	}
+
+	return cmd
+}
+
+func cleanVpcPeerGenericCluster(cmd *cobra.Command, args []string) {
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/generic.go]:
```suggestion
func cleanupVpcPeerGenericCluster(cmd *cobra.Command, args []string) {
```

File: pkg/subctl/cmd/cloud/vpcpeering/generic.go [hunk 12]
```diff
@@ -34,3 +34,18 @@ func newGenericVPCPeeringCommand() *cobra.Command {
 func vpcPeerGenericCluster(cmd *cobra.Command, args []string) {
 	// Nothing to peer here
 }
+
+func newCleanGenericVPCPeeringCommand() *cobra.Command {
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/generic.go]:
```suggestion
func newCleanupGenericVPCPeeringCommand() *cobra.Command {
```

File: pkg/subctl/cmd/cloud/vpcpeering/gcp.go [hunk 13]
```diff
@@ -42,24 +39,65 @@ func newGCPVPCPeeringCommand() *cobra.Command {
...
+		func(cloud api.Cloud, gwDeployer api.GatewayDeployer, reporter api.Reporter) error {
+			return errors.Wrap(err, cloud.CreateVpcPeering(cloudpreparegcp.NewCloud(*cloudInfo), reporter).Error())
+		})
+	if err != nil {
+		exit.OnErrorWithMessage(err, "Failed to create VPC Peering on GCP cloud")
+	}
+}
+
+// newCleanAWSVPCPeeringCommand removes a VPC Peering between different AWS clusters.
+func newCleanGCPVPCPeeringCommand() *cobra.Command {
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/gcp.go]:
```suggestion
func newCleanupGCPVPCPeeringCommand() *cobra.Command {
```

File: pkg/subctl/cmd/cloud/vpcpeering/aws.go [hunk 14]
```diff
@@ -42,27 +41,66 @@ func newAWSVPCPeeringCommand() *cobra.Command {
...
+	}
+
+	aws.ClientArgs.AddAWSFlags(cmd)
+	targetArgs.AddAWSFlags(cmd)
+
+	return cmd
+}
+
+// cleanVpcPeerAws removes peering object and routes between two OCP clusters in AWS.
+func cleanVpcPeerAws(cmd *cobra.Command, args []string) {
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/aws.go]:
```suggestion
func cleanupVpcPeerAws(cmd *cobra.Command, args []string) {
```

File: pkg/subctl/cmd/cloud/vpcpeering/aws.go [hunk 15]
```diff
@@ -42,27 +41,66 @@ func newAWSVPCPeeringCommand() *cobra.Command {
...
+		Run:   cleanVpcPeerAws,
+	}
+
+	aws.ClientArgs.AddAWSFlags(cmd)
+	targetArgs.AddAWSFlags(cmd)
+
+	return cmd
+}
+
+// cleanVpcPeerAws removes peering object and routes between two OCP clusters in AWS.
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/aws.go]:
```suggestion
// cleanupVpcPeerAws removes peering object and routes between two OCP clusters in AWS.
```

File: pkg/subctl/cmd/cloud/vpcpeering/aws.go [hunk 16]
```diff
@@ -42,27 +41,66 @@ func newAWSVPCPeeringCommand() *cobra.Command {
...
 	}
 }
+
+// newCleanAWSVPCPeeringCommand removes a VPC Peering between different AWS clusters.
+func newCleanAWSVPCPeeringCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "aws",
+		Short: "Removes VPC Peering on AWS cloud",
+		Long:  "This command cleans an OpenShift installer-provisioned infrastructure (IPI) on AWS cloud for Submariner uninstallation.",
+		Run:   cleanVpcPeerAws,
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/aws.go]:
```suggestion
		Run:   cleanupVpcPeerAws,
```

File: pkg/subctl/cmd/cloud/vpcpeering/aws.go [hunk 17]
```diff
@@ -42,27 +41,66 @@ func newAWSVPCPeeringCommand() *cobra.Command {
...
-			return cloud.CreateVpcPeering(targetCloud, reporter)
+			return errors.Wrap(err, cloud.CreateVpcPeering(targetCloud, reporter).Error())
 		})
 	if err != nil {
 		exit.OnErrorWithMessage(err, "Failed to create VPC Peering on AWS cloud")
 	}
 }
+
+// newCleanAWSVPCPeeringCommand removes a VPC Peering between different AWS clusters.
+func newCleanAWSVPCPeeringCommand() *cobra.Command {
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/aws.go]:
```suggestion
func newCleanupAWSVPCPeeringCommand() *cobra.Command {
```

File: pkg/subctl/cmd/cloud/vpcpeering/aws.go [hunk 18]
```diff
@@ -42,27 +41,66 @@ func newAWSVPCPeeringCommand() *cobra.Command {
...
 		func(cloud api.Cloud, gwDeployer api.GatewayDeployer, reporter api.Reporter) error {
-			return cloud.CreateVpcPeering(targetCloud, reporter)
+			return errors.Wrap(err, cloud.CreateVpcPeering(targetCloud, reporter).Error())
 		})
 	if err != nil {
 		exit.OnErrorWithMessage(err, "Failed to create VPC Peering on AWS cloud")
 	}
 }
+
+// newCleanAWSVPCPeeringCommand removes a VPC Peering between different AWS clusters.
```
Comment (ruromero) [inline pkg/subctl/cmd/cloud/vpcpeering/aws.go]:
```suggestion
// newCleanupAWSVPCPeeringCommand removes a VPC Peering between different AWS clusters.
```

**Review by ruromero — CHANGES_REQUESTED** [review]
I think you should use `cleanup` instead of just `clean` for the command and methods.

**Comment by r2dedios** [comment]
I'm Agree. "Clean" replaced by "Cleanup" as requested

**Pull request merged**